sklearn/neural_network/_multilayer_perceptron.py

```diff
@@ -702,7 +702,15 @@ def _fit_stochastic(
     def _update_no_improvement_count(self, early_stopping, X_val, y_val):
         if early_stopping:
             # compute validation score, use that for stopping
-            self.validation_scores_.append(self._score(X_val, y_val))
+            try:
+                val_score = self._score(X_val, y_val)
+            except ValueError as e:
+                if str(e) == "Input contains NaN.":
+                    val_score = np.inf
+                else:
+                    raise e
```

Comment:
This is quite brittle. It depends on the internals and the actual text raised by `accuracy_score` and `r2_score` (which are both not ideal lol)

I rather have a check here explicitly for `nan`, and if we have `nan`, do the logic, instead of relying on the scorer to raise that exception.

Reply:
X_val and y_val don't have any NaN. But because they contain really large numbers, the computation of the score generates NaN values, which, when validation runs, raises the error.

The exception is raised deep in the code (check #29504 for the full traceback) so that's why I ended up just adding the try: except: statement.

Reply:
ok, I did manage to do it better :) thank you for pushing me